VideoLocker/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -60,4 +73,142 @@ public static void normalizeCourseStructure(CourseStructureV1Model courseStructu
+                    items.add( model );
+                }
+            }
+        }
+        return items;
+    }
+
+    /**
+     * from new VideoBlockModel to legacy VideoRsponseModel
+     * @param videoBlockModel
+     * @return
+     */
+    public static VideoResponseModel mappingVideoResponseModelFrom(VideoBlockModel videoBlockModel){
+        VideoResponseModel model = new VideoResponseModel();
+        model.setCourseId(videoBlockModel.getRoot().getId());
+        SummaryModel summaryModel = mappingSummaryModelFrom(videoBlockModel);
+        model.setSummary(summaryModel);
+        model.videoBlockModel = videoBlockModel;
+        model.setSectionUrl(videoBlockModel.getParent().getBlockUrl());
+        model.setUnitUrl(videoBlockModel.getBlockUrl());
+
+        return model;
+    }
+
+
+    private static SummaryModel mappingSummaryModelFrom(VideoBlockModel videoBlockModel){
+        SummaryModel model = new SummaryModel();
+        model.setType(videoBlockModel.getType());
+        model.setDisplayName(videoBlockModel.getDisplayName());
+        model.setVideoUrl(videoBlockModel.getData().encodedVideos.getPreferredVideoInfo().url);
+        model.setDuration((int)videoBlockModel.getData().duration);
+        //FIXME - is this field missing?
+        //model.setOnlyOnWeb(videoBlockModel.);
+       // public boolean onlyOnWeb;
+        model.setId(videoBlockModel.getId());
+        model.setSize(videoBlockModel.getSize());
+        model.setTranscripts(videoBlockModel.getData().transcripts);
+        //FIXME = is this field missing?
+       // private EncodingsModel encodings;
+        return model;
+    }
+
+    /**
+     * from new CourseComponent to legacy data structure.
+     * @param courseComponent
+     * @return
+     */
+    public static Map<String, SectionEntry> mappingCourseHierarchyFrom(CourseComponent courseComponent){
+        Map<String, SectionEntry> map = new HashMap<>();
+        for(IBlock block : courseComponent.getChildren()){
+            CourseComponent chapter = (CourseComponent)block;
+            SectionEntry entry = new SectionEntry();
+            entry.chapter = chapter.getName();
+            entry.isChapter = true;
+            entry.section_url = chapter.getBlockUrl();
+            map.put(entry.chapter, entry);
+
+            for( IBlock subBlock : chapter.getChildren() ){
+                CourseComponent section = (CourseComponent)subBlock;
+
+                entry.sections.put(section.getName(),
+                    (ArrayList)CourseManager.mappingAllVideoResponseModelFrom(section, null));
+            }
+        }
+        return map;
+    }
+
+
+    public static LectureModel getLecture(CourseComponent courseComponent, String chapterName, String lectureName)
+        throws Exception {
+        //TODO - we may use a generic filter to fetch the data?
+        for(IBlock chapter : courseComponent.getChildren()){
+            if ( chapter.getDisplayName().equals(chapterName) ){
```

Comment:
@aleffert  yeah, we should use id instead,  but I need to figure out how to  make the code backward compatible. Let me do some study